src/lib.rs

```diff
@@ -109,6 +109,44 @@ pub use descriptor::Descriptor;
 pub use miniscript::Miniscript;
 pub use miniscript::satisfy::{BitcoinSig, Satisfier};
 
+///Trait for converting to Hash160 type required for encoding script into PkH
+pub trait ToHash160 {
+    fn to_hash160(&self) -> hash160::Hash;
+}
+
+impl ToHash160 for hash160::Hash{
+
+    fn to_hash160(&self) -> hash160::Hash{
+        *self
+    }
+}
+///Public key trait which can be converted to Hash type
+pub trait MiniscriptKey: Clone + Eq + Ord + str::FromStr + fmt::Debug +fmt::Display {
+    type Hash: Clone + Eq + Ord + str::FromStr + fmt::Display + fmt::Debug + ToHash160 + hash::Hash;
```

Comment:
`ToHash160` shouldn't be one of these bounds; it prevents use of stringly-typed hashes.

Reply:
How should I encode the to `Script` without that bound?

Reply:
encoding to `Script` already requires the `ToPublicKey` bound. Add a `Pk::Hash: ToHash160` bound as well I guess